tests/rptest/tests/consumer_offsets_migration_test.py

```diff
@@ -0,0 +1,123 @@
+    @parametrize(failures=True)
+    @parametrize(failures=False)
+    def test_migrating_consume_offsets(self, failures):
+        '''
+        Validates correctness while executing consumer offsets migration
+        '''
+
+        # set redpanda logical version to value without __consumer_offsets support
+        self.redpanda = RedpandaService(
+            self.test_context,
+            5,
+            extra_rp_conf={
+                "group_topic_partitions": 16,
+                "default_topic_replications": 3,
+            },
+            environment={"__REDPANDA_LOGICAL_VERSION": 1})
+
+        self.redpanda.start()
+        self._client = DefaultClient(self.redpanda)
+        # set of failure suppressed nodes - required to make restarts deterministic
+        suppressed = set()
+
+        def failure_injector_loop():
+            f_injector = FailureInjector(self.redpanda)
+            while failures:
+                f_type = random.choice(FailureSpec.FAILURE_TYPES)
+                length = 0
+                node = random.choice(self.redpanda.nodes)
+                while self.redpanda.idx(node) in suppressed:
+                    node = random.choice(self.redpanda.nodes)
+
+                # allow suspending any node
+                if f_type == FailureSpec.FAILURE_SUSPEND:
+                    length = random.randint(
+                        1,
+                        ConsumerOffsetsMigrationTest.max_suspend_duration_sec)
+
+                f_injector.inject_failure(
+                    FailureSpec(node=node, type=f_type, length=length))
+
+                delay = random.randint(
+                    ConsumerOffsetsMigrationTest.min_inter_failure_time_sec,
+                    ConsumerOffsetsMigrationTest.max_inter_failure_time_sec)
+                self.redpanda.logger.info(
+                    f"waiting {delay} seconds before next failure")
+                time.sleep(delay)
+
+        if failures:
+            finjector_thread = threading.Thread(target=failure_injector_loop,
+                                                args=())
+            finjector_thread.daemon = True
+            finjector_thread.start()
+        spec = TopicSpec(partition_count=6, replication_factor=3)
+        self.client().create_topic(spec)
+        self.topic = spec.name
+
+        self.start_producer(1, throughput=5000)
+        self.start_consumer(1)
+        self.await_startup()
+
+        def cluster_is_stable():
+            admin = Admin(self.redpanda)
+            brokers = admin.get_brokers()
+            if len(brokers) < 3:
+                return False
+
+            for b in brokers:
+                self.logger.debug(f"broker:  {b}")
+                if not (b['is_alive'] and 'disk_space' in b):
+                    return False
+
+            return True
+
+        # enable consumer offsets support
+        self.redpanda.set_environment({"__REDPANDA_LOGICAL_VERSION": 2})
```

Comment:
Would be good to assert that __consumer_offsets is not present before the start of the test, and that the groups topic exists in a non-empty state.

Reply:
good point, i will add that

Reply:
we can not ask about `kafka_internal` topic offsets directly, i will query for consumer groups to prove it exists.